pyccel/codegen/pipeline.py

```diff
@@ -228,6 +287,26 @@ def get_module_dependencies(parser, mods=(), folders=()):
 
         dep_mods, inc_dirs = get_module_dependencies(parser)
 
+        # Determine if the module or program requires internal libraries
+        # by iterating over internal_libs list that containes internal libs
+        # implemented inside pyccel and compare them by module includes
+        # returned by printer includes through get_additional_imports function
+        for lib in internal_libs:
+            if lib in codegen.get_printer_imports():
+                # get the include folder path and library files
+                if lib not in internal_lib_dict:
+                    internal_lib_dict[lib] = get_internal_lib(lib)
```

Comment:
It looks like you have moved this to after `convert_only` to avoid compiling when the user asked you not to. However `convert_only` also means that you need to generate all necessary code. So I think the libraries should still be collected and copied before the `if convert_only: continue`. I would use `get_internal_lib` to get the internal libraries and then add them to a list of files to be compiled which can all use the same method for compiling